packages/runtime/src/widgets/Form/TextInput.tsx

```diff
@@ -145,8 +145,19 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
 
     if (values?.validator?.regex) {
       rulesArray.push({
-        pattern: new RegExp(values.validator.regex),
-        message: values.validator.regexError || "The field has invalid value",
+        validator: (_, inputValue?: string) => {
+          if (
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
You shouldn't need this. Typescript should know the type is not undefined/null if you check for it in the if guard

Reply:
it still complains
![image](https://github.com/user-attachments/assets/25032f9b-29c0-434e-a8e6-db1a9cf4e774)

Reply:
Assign it to a variable and you should be good, e.g.

```ts
    const regex = values?.validator?.regex;
    if (regex) {
```